Add flag to specify alternative default git ref

nweedo3 opened the pull request:
Add the `--git-default-chart-ref` flag which enables the modification
of the ref used to clone a chart if the `.spec.git.ref` field is
unspecified. The original default of `master` is maintained.

Related issue: https://github.com/fluxcd/helm-operator/issues/80

<!--
# General contribution criteria

Please have a look at our contribution guidelines: https://github.com/fluxcd/helm-operator/blob/master/CONTRIBUTING.md
Particularly the sections about the:

 - DCO;
 - contribution workflow; and
 - how to get your fix accepted

To help the maintainers out when they're writing release notes, please
try to include a sentence or two here describing your change for end
users. See the CHANGELOG.md file in the top-level directory for examples.

Particularly for ground-breaking changes and new features, it's important to
make users and developers aware of what's changing and where those changes
were documented or discussed.

Even for smaller changes it's useful to see things documented as well, as it
gives everybody a chance to see at a glance what's coming up in the next
release. It makes the life of the project maintainer a lot easier as well.
-->

stefanprodan commented on chart/helm-operator/templates/deployment.yaml:
```diff
@@ -145,6 +145,7 @@ spec:
         {{- end }}
         - --git-timeout={{ .Values.git.timeout }}
         - --git-poll-interval={{ .Values.git.pollInterval }}
+        - --git-default-chart-ref={{ .Values.git.defaultChartRef }}
```
Wrap this in an `if` to maintain backwards compatibility:

```
{{- if .Values.git.defaultChartRef }}
- --git-default-chart-ref={{ .Values.git.defaultChartRef }}
{{- end }}
```

nweedo3 replied:
@stefanprodan sure. Would you prefer the entire PR in one commit, or are you fine with me adding the review changes on top of the original commit?

stefanprodan replied:
One commit is fine, amend your commit and force push it. Thanks

stefanprodan commented on pkg/apis/helm.fluxcd.io/v1/types.go:
```diff
@@ -5,7 +5,7 @@ import (
 
 	"github.com/ghodss/yaml"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Please remove this alias, it has no effect since it's the same with the package name.

nweedo3 commented:
@stefanprodan the changes should be good to go now - thanks for the review.

nweedo3 commented:
@stefanprodan do you have any further updates on this? Thanks.

stefanprodan reviewed (approved):
LGTM

hiddeco commented on cmd/helm-operator/main.go:
```diff
@@ -112,6 +113,7 @@ func init() {
 
 	gitTimeout = fs.Duration("git-timeout", 20*time.Second, "duration after which git operations time out")
 	gitPollInterval = fs.Duration("git-poll-interval", 5*time.Minute, "period on which to poll git chart sources for changes")
+	gitDefaultChartRef = fs.String("git-default-chart-ref", "master", "ref to clone chart from if ref is unspecified in a HelmRelease")
```
I think the length of the flag can be reduced to `--git-default-ref`, as the only Git thing the operator does is syncing charts.

nweedo3 committed:
Add flag to specify alternative default git ref

Add the `--git-default-ref` flag which enables the modification
of the ref used to clone a chart if the `.spec.git.ref` field is
unspecified. The original default of `master` is maintained.

Related issue: https://github.com/fluxcd/helm-operator/issues/80

hiddeco reviewed (approved):
Thank you @niall-weedon :1st_place_medal:

nweedo3 commented:
No problem @hiddeco, thanks for the review! :)
